scipy/stats/_sobol.pyi

```diff
@@ -0,0 +1,52 @@
+import numpy.typing as npt
+from scipy._lib._util import IntNumber
+
+def initialize_v(
+    v : npt.ArrayLike, 
```

Comment:
```suggestion
import numpy as np
from scipy._lib._util import IntNumber

def initialize_v(
    v : np.ndarray, 
```
@tupui I'm not sure I'm interpreting this correctly, but does this signature not mean that the function operates on an ndarray, and not arbitrary array-likes?
https://github.com/scipy/scipy/blob/4ec6f29c0344ddce80845df0c2b8765fc8f27a72/scipy/stats/_sobol.pyx#L188
If so, it seems that all instances of `npt.ArrayLike` will have to be replaced with `np.ndarray` throughout the file.

Reply:
Indeed, but then the current types in `_qmc_cy.pyi` are also not correct (this is even clearer there as the functions are calling things like `sample_view.shape[0]`). I guess it went through the crack for simplicity.

@akpandeya would you like to make the changes? It would be great if you could do it for `_qmc_cy.pyi` as well then. Let me know, I can also do this before merging.

Reply:
Sure thing! Will do it by tomorrow.

Reply:
I guess this needs to be addressed? If you want this PR to get merged for the 1.7.0 release try to get the change done soon, I will branch quite soon, targeting Wednesday